iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2371,6 +2377,35 @@ static STRING_HANDLE buildClientId(const char* device_id, const char* module_id)
     }
 }
 
+//
+// build the parameter and add to the username.
+//
+static int addUsernameParameter(STRING_HANDLE username, const char* token, const char* value)
```

Comment:
addUsernameProperties()?

Reply:
Code was going to start being duplicated. This is a refactor.  I'm up for naming suggestions.

Reply:
How about addUserNameProperty(). It only adds one at a time.